### ISSUE TITLE
Error Handling & status codes - default and route specific

### ISSUE THREAD

**Issue by elyrank** [softwaremill/tapir#85]
I'd like to have the 'default' error types defined once and to be able to handle exceptions and return those error types.

i.e. have one place that defines: 400, 401, 404, 409, etc. with a default message (i.e. 401 Not Authorized)
and if there is a specific error for a specific route - add its own error type and doc, and be able to override an already existing status code.
i.e. in the specific route - 401 "You don't have permissions to modify the book store"

**Comment by matwojcik** [comment]
I'm not sure if issue is about the same what lacks for me, so I'll shortly describe it here: I'd like to be able to document each error type separately if they e.g. change status code.

Currently:
```scala
  sealed trait ValidationFailure extends Product with Serializable
  case class OrderNotExists(id: Order.Id) extends ValidationFailure
  case class OrderNotFulfilled(id: Order.Id) extends ValidationFailure

endpoint.in("validate" / path[Int]("orderId").map(Order.Id(_))(_.value))
      .out(jsonBody[Order])
      .errorOut(
        statusFrom(
          jsonBody[ValidationFailure].description("validation failure").example(OrderNotFulfilled(Order.Id(3))),
          StatusCodes.PreconditionFailed,
          (whenClass[OrderNotExists], StatusCodes.NotFound)
        )
      )
```
in openapi both error status codes will be presented, however both have the same description, example etc.

**Comment by adamw** [comment]
Released in 0.8.0
